src/core/features/fileuploader/services/fileuploader-helper.ts

```diff
@@ -698,10 +701,34 @@ export class CoreFileUploaderHelperProvider {
     async uploadFileEntry(
         fileEntry: FileEntry,
         deleteAfter: boolean,
-        maxSize?: number,
+        maxSize: number | undefined,
+        upload: true,
+        allowOffline?: boolean,
+        name?: string,
+    ): Promise<CoreWSUploadFileResult>;
+    async uploadFileEntry(
+        fileEntry: FileEntry,
+        deleteAfter: boolean,
+        maxSize: number | undefined,
+        upload: false,
+        allowOffline?: boolean,
+        name?: string,
+    ): Promise<FileEntry>;
+    async uploadFileEntry(
+        fileEntry: FileEntry,
+        deleteAfter: boolean,
+        maxSize: number | undefined,
         upload?: boolean,
         allowOffline?: boolean,
         name?: string,
+    ): Promise<FileEntry>;
```

Comment:
If I'm not mistaken, this one should be:

```suggestion
    ): Promise<CoreWSUploadFileResult | FileEntry>;
```